src/components/input/DurationInput.js

```diff
@@ -0,0 +1,415 @@
+        color={inverse ? 'neutral' : 'teal'}
+        tint={inverse ? 'lightest' : 'darkest'}
+        flex="0 0 18px"
+        paddingHorizontal={2}
+      >
+        :
+      </TextBody>
+    );
+  };
+
+  splitSecondsIntoHoursMinutesSeconds = seconds => {
+    const secondsInMinute = 60;
+    const secondsInHour = secondsInMinute * 60;
+
+    const hours = Math.floor(seconds / secondsInHour);
+    const minutes = Math.floor((seconds % secondsInHour) / secondsInMinute);
+    const remainingSeconds = seconds % secondsInMinute;
+    return {
+      hours,
+      minutes,
+      seconds: remainingSeconds,
+    };
+  };
+
+  render() {
+    const {
+      className,
+      commonInputProps,
+      hoursInputProps,
+      inverse,
+      minutesInputProps,
+      onFocus,
+      onBlur,
+      secondsInputProps,
+      size,
+      ...others
+    } = this.props;
+
+    const classNamesHours = cx(
+      theme['wrapper'],
+      {
+        [theme['has-focus']]: this.state.hoursInputHasfocus,
+        [theme['has-error']]: hoursInputProps.error,
+        [theme['has-warning']]: hoursInputProps.warning,
+        [theme['has-success']]: hoursInputProps.success,
+        [theme['is-disabled']]: commonInputProps.disabled,
+        [theme['is-inverse']]: inverse,
+        [theme['is-read-only']]: commonInputProps.readOnly,
+      },
+      className,
+    );
+
+    const classNamesMinutes = cx(
+      theme['wrapper'],
+      {
+        [theme['has-focus']]: this.state.minutesInputHasfocus,
+        [theme['has-error']]: minutesInputProps.error,
+        [theme['has-warning']]: minutesInputProps.warning,
+        [theme['has-success']]: minutesInputProps.success,
+        [theme['is-disabled']]: commonInputProps.disabled,
+        [theme['is-inverse']]: inverse,
+        [theme['is-read-only']]: commonInputProps.readOnly,
+      },
+      className,
+    );
+
+    const classNamesSeconds = cx(
+      theme['wrapper'],
+      {
+        [theme['has-focus']]: this.state.secondsInputHasfocus,
+        [theme['has-error']]: secondsInputProps.error,
+        [theme['has-warning']]: secondsInputProps.warning,
+        [theme['has-success']]: secondsInputProps.success,
+        [theme['is-disabled']]: commonInputProps.disabled,
+        [theme['is-inverse']]: inverse,
+        [theme['is-read-only']]: commonInputProps.readOnly,
+      },
+      className,
+    );
+
+    const boxProps = pickBoxProps(others);
+
+    const splitSecondsIntoHoursMinutesSeconds = seconds => {
+      const secondsInMinute = 60;
+      const secondsInHour = secondsInMinute * 60;
+  
+      const hours = Math.floor(seconds / secondsInHour);
+      const minutes = Math.floor((seconds % secondsInHour) / secondsInMinute);
+      const remainingSeconds = seconds % secondsInMinute;
+      return {
+        hours,
+        minutes,
+        seconds: remainingSeconds,
+      };
+    };
+
+    const { hours, minutes, seconds } = splitSecondsIntoHoursMinutesSeconds(this.state.seconds);
```

Comment:
Now you're still calling the function defined within render?

Reply:
I moved the function out of the component and placed it in the utils folder. So we can use it in other files if necessary.